app/main/views/g_cloud.py

```diff
@@ -340,54 +332,77 @@ def save_search(framework_framework):
         **build_search_query(request.args, filters.values(), content_manifest, lots_by_slug)
     )
 
+    projects = get_direct_award_projects(current_user.id, 'open_projects', 'name')
+
     return render_template(
         'direct-award/save-search.html',
         framework_framework=framework_framework,
         search_summary_sentence=search_summary.markup(),
         search_api_url=search_api_url,
         form=CreateProjectForm(),
+        projects=projects,
     )
 
 
 @direct_award.route('/<string:framework_framework>/projects/create', methods=['POST'])
 def project_create(framework_framework):
     form = CreateProjectForm()
-    if form.validate_on_submit():
-        name = form.name.data
+    name = form.name.data
+
+    try:
+        save_search_selection = request.form['save_search_selection']
+    except:
+        save_search_selection = None
+
+    name_is_invalid = save_search_selection == "new_search" and not name
+
+    if not form.validate_on_submit() or not save_search_selection or name_is_invalid:
+        projects = get_direct_award_projects(current_user.id, 'open_projects', 'name')
+        projects.sort(key=lambda x: x['name'])
+        frameworks_by_slug = framework_helpers.get_frameworks_by_slug(data_api_client)
+        search_meta = SearchMeta(request.form['search_api_url'], frameworks_by_slug)
+
+        if not name:
+            form.name.errors = ["Names must be between 1 and 100 characters"]
```

Comment:
Presumably this string comes from somewhere in the forms system, and it'd have been vaguely nice to re-use that rather than hard-code it here, but I guess let's not worry for now.

Reply:
I couldn't find a way to force it to throw an error. If you know a way I'd be happy to try it out